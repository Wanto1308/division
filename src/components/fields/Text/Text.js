import React from 'react';
import PropTypes from 'prop-types';
import IconButton from '../../elements/IconButton';
import styles from './styles.scoped.css';

export default function Text(props) {
  const { className, input, inputProps, onClickIcon, meta } = props;
  const { active, dirty, error, touched, asyncValidating } = meta;
  const classes = [
    styles.root,
    active && styles.focus,
    inputProps.required && styles.required,
    !!error && (dirty || touched) && styles.error,
    className
  ].filter(Boolean).join(' ');

  return (
    <div className={classes}>
      <label htmlFor={input.name}>
        {inputProps.label}
        {inputProps.required && <span>*</span>}
      </label>
      <input id={input.name} {...input} {...inputProps} />
      {inputProps.icon && (<IconButton disabled={inputProps.disabled} icon={inputProps.icon}
        onClick={onClickIcon} size={20}/>)}
      {!!error && (dirty || touched) && <small>{error}</small>}
      {asyncValidating && <small>Sedang Mengecek...</small>}
    </div>
  );
}

Text.defaultProps = {
  className: '',
  input: {},
  inputProps: {},
  meta: {},
  onClickIcon: () => {},
};

Text.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  meta: PropTypes.object,
  onClickIcon: PropTypes.func,
};
