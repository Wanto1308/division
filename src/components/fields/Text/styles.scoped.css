.root {
  align-items: center;
  display: grid;
  grid-template-columns: 1fr auto;
  padding-bottom: 1.75rem;
  position: relative;
}

.root:after {
  background-color: var(--black-color-4);
  border-radius: 0.0625rem;
  bottom: 1.75rem;
  content: '';
  height: 0.0625rem;
  left: 0;
  position: absolute;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  width: 100%;
}

.root > input {
  background: none;
  border: none;
  color: var(--black-color-1);
  font-size: 1rem;
  line-height: 1.1875rem;
  padding: 0.375rem 0.5rem 0.5rem;
  width: 100%;
}

.root > input::placeholder {
  color: var(--black-color-5);
}

.root > input:disabled,
.root > input:disabled::placeholder {
  color: var(--black-color-3);
}

.root > input:read-only {
  cursor: default;
}

.root > button {
  margin-right: 0.375rem;
}

.root > label {
  color: var(--black-color-3);
  display: block;
  grid-column: 1 /span 2;
  padding: 0.5rem 0.5rem 0;
  width: fit-content;
  white-space: nowrap;
}

.root > small {
  color: var(--red-color);
  left: 0.5rem;
  position: absolute;
  top: 3.75rem;
  text-align: left;
}

.root.focus::after {
  background-color: var(--primary-color);
}

.root.error::after {
  background-color: var(--red-color);
}

.root.required > label > span {
  color: var(--red-color);
}
