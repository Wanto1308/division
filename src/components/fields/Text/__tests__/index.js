import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Text from '../Text';

describe('src/components/fields/Text', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Text />);
    Text.defaultProps.onClickIcon();
    expect(tree).toMatchSnapshot();
  });

  test('error', () => {
    const meta = { active: true, error: 'err', dirty: false, touched: true };

    const result = Text({ ...Text.defaultProps, meta });
    expect(result.props.children[3].props.children).toBe('err');
  });

  test('required', () => {
    const inputProps = { required: true };

    const result = Text({ ...Text.defaultProps, inputProps });
    expect(result.props.children[0].props.children[1].props.children).toBe('*');
  });

  test('with icon', () => {
    const inputProps = { icon: 'test' };

    const result = Text({ ...Text.defaultProps, inputProps });
    expect(result.props.children[2].props.icon).toBe(inputProps.icon);
  });
});
