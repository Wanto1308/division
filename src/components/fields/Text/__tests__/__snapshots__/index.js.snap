// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Text render 1`] = `
<div
  className=""
>
  <label />
  <input />
</div>
`;
